cmd/migration/main.go

```diff
@@ -0,0 +1,209 @@
+package main
+
+import (
+	"encoding/json"
+	"flag"
+	"fmt"
+	"os"
+
+	"github.com/automationbroker/bundle-lib/apb"
+	"github.com/automationbroker/bundle-lib/clients"
+	crd "github.com/openshift/ansible-service-broker/pkg/dao/crd"
+	etcd "github.com/openshift/ansible-service-broker/pkg/dao/etcd"
+	"github.com/sirupsen/logrus"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var options struct {
+	EtcdHost           string
+	EtcdCAFile         string
+	EtcdClientCert     string
+	EtcdClientKey      string
+	EtcdPort           int
+	MigrationNamespace string
+}
+
+func init() {
+	flag.IntVar(&options.EtcdPort, "port", 2379, "user '--port' option to specify the port to connect to etcd")
+	flag.StringVar(&options.EtcdHost, "host", "", "host to connect to etcd server")
+	flag.StringVar(&options.EtcdCAFile, "ca-file", "", "CA certificate file path to be used to connect over TLS with etcd server")
+	flag.StringVar(&options.EtcdClientCert, "client-cert", "", "client cert file path to authenticate to etcd server. If used must also use client-key")
+	flag.StringVar(&options.EtcdClientKey, "client-key", "", "client key file path to authenticate to etcd server. If used must also use client-cert")
+	flag.StringVar(&options.MigrationNamespace, "namespace", "", "namepsace that the migration should run in")
+	flag.Parse()
+}
+
+var crdDao *crd.Dao
+var etcdDao *etcd.Dao
+
+func main() {
+	con := clients.EtcdConfig{
+		EtcdHost:       options.EtcdHost,
+		EtcdPort:       options.EtcdPort,
+		EtcdCaFile:     options.EtcdCAFile,
+		EtcdClientKey:  options.EtcdClientKey,
+		EtcdClientCert: options.EtcdClientCert,
+	}
+	logrus.SetLevel(logrus.DebugLevel)
+	clients.InitEtcdConfig(con)
+	logrus.Infof("etcd configuration: %v", con)
+	var err error
+	etcdDao, err = etcd.NewDao()
+	if err != nil {
+		panic(fmt.Sprintf("Unable to connect to etcd - %v", err))
+	}
+	crdDao, err = crd.NewDao(options.MigrationNamespace)
+	if err != nil {
+		panic(fmt.Sprintf("Unable to connect to crd - %v", err))
```

Comment:
NIT: I don't think this connects to the crd resource.  It only creates the client.   Maybe a different error message should be here.

Reply:
Waiting on more reviews before updating this one

Reply:
@shawn-hurley might also be worth a precheck on the bundle resource. Maybe something like this: 
```go
_, err := resources.Bundle.Bundles(c.config.Namespace).List(metav1.ListOptions{})
if err != nil {
....
}
```

Reply:
If the bundles already exist it will error and the migration will fail, I think it would be better to attempt to migrate IMO.